### ISSUE TITLE
Add full component string name support

### ISSUE THREAD

**Issue by ayebear** [Lastresort92/OCS#10]
Add support for removing, adding, assigning, etc. components using their bound string name. This will allow for better networking/file support.

Note: Could also add the ability to get components with their name, but it would have to be the base Component class reference, because you can't just change types in code on runtime.

**Comment by kevinwmiller** [comment]
Do you mean in addition to addComponent<Position>(Position(0, 0)), you want something like
addComponent("Position", Position(0, 0))?

**Comment by ayebear** [comment]
No, it wouldn't require the type to be used. Basically, you'd only be able to use the base component class methods, like serialize and deSerialize, because you don't know the type. So, you could do things like this:

```
objects.addComponents(id, "Position", "0 0");
```

Also, check out issue #9 first to simplify this a bit more.

**Comment by kevinwmiller** [comment]
Gotchya, Seems like a nice way to initialize from a file.

**Comment by kevinwmiller** [comment]
Just to clarify, the name is the name of the component and not the object (however, will add this feature as well) right?

**Comment by ayebear** [comment]
Correct, it's the name of the component type, like "Position".
